go/consensus/tendermint/light/service.go

```diff
@@ -0,0 +1,337 @@
+		return
+	case <-c.ctx.Done():
+		return
+	case <-c.consensus.Synced():
+	}
+
+	// Store latest light block into the trusted store.
+	lb, err := c.consensus.GetLightBlock(c.ctx, consensusAPI.HeightLatest)
+	if err != nil {
+		c.logger.Error("failed to obtain latest light block from consensus", "err", err)
+		return
+	}
+	tmLb, err := c.consensusBlkToTendermint(lb)
+	if err != nil {
+		c.logger.Error("failed to convert consensus light block to tendermint", "err", err)
+		return
+	}
+	if err = c.store.SaveLightBlock(tmLb); err != nil {
+		c.logger.Error("failed to initialize light client store with a trusted block", "err", err)
+		return
+	}
+
+	chainCtx, err := c.consensus.GetChainContext(c.ctx)
+	if err != nil {
+		c.logger.Error("failed to obtain chain context", "err", err)
+		return
+	}
+	tmGenDoc, err := tmapi.GetTendermintGenesisDocument(c.genesis)
+	if err != nil {
+		c.logger.Error("failed to obtain tendermint genesis document", "err", err)
+		return
+	}
+
+	// Initialize a provider pool.
+	pool := p2pLight.NewLightClientProviderPool(c.ctx, chainCtx, tmGenDoc.ChainID, c.p2p)
+	var providers []tmlightprovider.Provider
+	for i := 0; i < numProviders; i++ {
+		p := pool.NewLightClientProvider()
+		providers = append(providers, p)
+		c.providers = append(c.providers, p)
+	}
+	tmc, err := tmlight.NewClientFromTrustedStore(
+		tmGenDoc.ChainID,
+		viper.GetDuration(CfgConsensusLightClientTrustPeriod),
+		providers[0],  // Primary provider.
+		providers[1:], // Witnesses.
+		c.store,
+		tmlight.MaxRetryAttempts(lcMaxRetryAttempts),
+		tmlight.Logger(common.NewLogAdapter(!viper.GetBool(common.CfgLogDebug))),
+		tmlight.DisableProviderRemoval(),
+	)
+	if err != nil {
+		c.logger.Error("failed to obtain initialize tendermint light client", "err", err)
+		return
+	}
+	c.lc = &lightClient{tmc: tmc}
+	close(c.initCh)
+	initOk = true
+
+	// Watch epochs and insert new trusted blocks on every epoch transition.
+	ch, sub, err := c.consensus.Beacon().WatchEpochs(c.ctx)
+	if err != nil {
+		c.logger.Error("failed to obtain watch epochs", "err", err)
+		return
+	}
+	defer sub.Close()
+	for {
+		select {
+		case <-c.stopCh:
+			return
+		case <-c.ctx.Done():
+			return
+		case <-ch:
+			// Store latest light block into the trusted store.
+			lb, err := c.consensus.GetLightBlock(c.ctx, consensusAPI.HeightLatest)
+			if err != nil {
+				c.logger.Error("failed to obtain latest light block from consensus", "err", err)
+				return
+			}
+			tmLb, err := c.consensusBlkToTendermint(lb)
+			if err != nil {
+				c.logger.Error("failed to convert consensus light block to tendermint", "err", err)
+				continue
+			}
+			if err = c.store.SaveLightBlock(tmLb); err != nil {
+				c.logger.Error("failed to initialize light client store with a trusted block", "err", err)
+				continue
+			}
+		}
+	}
+}
+
+// GetLightBlock implements api.Client.
+func (c *client) GetLightBlock(ctx context.Context, height int64) (*consensus.LightBlock, rpc.PeerFeedback, error) {
```

Comment:
Given that this particular light client service also has a full node available, should it always first try to fetch the block from the consensus backend? This way the callers don't need to worry about querying the consensus backend first and can just always invoke the light client.